services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -806,21 +864,34 @@ private boolean isCentecSwitch(IOFSwitch sw) {
         return StringUtils.contains(sw.getSwitchDescription().getManufacturerDescription(), "Centec");
     }
 
-    private void buildAndInstallMeter(IOFSwitch sw, Set<OFMeterFlags> flags, long bandwidth, long burstSize,
-                                      long meterId) throws OfInstallException {
-        logger.info("Installing meter {} on switch {} with bandwidth {}", meterId, sw.getId(), bandwidth);
+    private void buildAndInstallMeter(OFMeterMod meterMod, IOFSwitch sw, long bandwidth) throws OfInstallException {
+        logger.info("Installing meter {} on switch {} with bandwidth {}", meterMod.getMeterId(), sw.getId(), bandwidth);
 
+        pushFlow(sw, "--InstallMeter--", meterMod);
```

Comment:
I meant that buildAndInstallMeter() and buildAndModifyMeter() must use buildMeterMode() inside themselfes. To do not duplicate meterMode creation, but also do not lose incapsulation of buildAndInstallMeter() and buildAndModifyMeter() methods.

Something like 

```
private void buildAndInstallMeter(IOFSwitch sw, Set<OFMeterFlags> flags, long bandwidth,
     long burstSize, long meterId) throws OfInstallException {	        
***
meterMode = buildMeterMode(***)
pushFlow(sw, "--InstallMeter--", meterMod);


```